### ISSUE TITLE
[A11Y] Tag has role="button" and aria-label="button" when it's not clickable

### ISSUE THREAD

**Issue by VladimirMilenko** [uber-web/baseui#1788]
`Tag` has role="button" and aria-label="button" when it's not clickable, this violates the A11y rules :)

https://codesandbox.io/s/gz23x?module=/example.js

## Current Behavior
`Tag` has role="button" and aria-label="button"

## Expected Behavior
`Tag` shouldn't have a role and aria-label

| Tech    | Version |
| ------- | ------- |
| Base UI | v8.16.5 |

- [x] I have searched the [issues](https://github.com/uber-web/baseui/issues) of this repository and believe that this is not a duplicate.

**Comment by VladimirMilenko** [comment]
As a workaround I'm still able to set this via:
```
const TitleTagOverrides = {
  Root: {
    style: {
      marginTop: 0,
      marginBottom: 0,
      marginLeft: 0,
      marginRight: 0
    },
    props: {
      role: null,
      "aria-label": null
    }
  }
};
```

**Comment by gergelyke** [comment]
@VladimirMilenko thanks for the report! any chance you have the time to put together a PR fixing it?

**Comment by rohit012** [comment]
@gergelyke Working on the issue, will share a PR ASAP.

**Comment by Shubhamchinda** [comment]
Hey, can I take this ?

**Comment by gergelyke** [comment]
@Shubhamchinda - sorry, @rohit012 just fixed it :) there are a few more help wanted issues, if you are interested!

**Comment by Shubhamchinda** [comment]
> @Shubhamchinda - sorry, @rohit012 just fixed it :) there are a few more help wanted issues, if you are interested!

Yeah, I missed it. Sorry!
I'll be around if something comes up! :)
